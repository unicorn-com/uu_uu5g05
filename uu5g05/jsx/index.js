// used from libraries depending on uu5g05 (this gets bundled into them); used due to uu5devkitg01-plugin's .babelrc config regarding @babel/preset-react preset
const Uu5 = require("uu5g05");
// the fallback is for temporary flow as described in ./jsx-runtime.js
Object.assign(exports, Uu5._jsx || { createElement: Uu5.Utils.Element.create });
