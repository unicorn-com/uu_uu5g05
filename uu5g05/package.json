{
  "name": "uu5g05",
  "version": "1.40.1",
  "description": "uu5 Library contains basic hooks, providers, components, utils and environment for easy development of uuApps or uuLibs.",
  "license": "License Commercial",
  "namespace": "Uu5",
  "spec": "uu5-lib; uu_appg01_devkit@6.6.0; uu5devkitg01-plugin@1.5.1",
  "dependencies": {
    "prop-types": "15.7.2",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "regenerator-runtime": "0.13.7",
    "uu5stringg01": "^1.10.0"
  },
  "uuBuildSettings": {
    "externals": {
      "react": true,
      "react-dom": true,
      "prop-types": true
    }
  },
  "devDependencies": {
    "@emotion/css": "11.9.0",
    "@mdi/font": "7.0.96",
    "memoize-one": "5.1.1",
    "systemjs": "0.19.47",
    "uu_appg01_devkit": "^6.6.0",
    "uu_appg01_oidc": "^4.0.0",
    "uu_datatypesg01": "^0.1.1",
    "uu5devkitg01-plugin": "^1.5.1",
    "uu5g05-test": "*",
    "uu5loaderg01": "^1.7.0"
  },
  "engines": {
    "node": ">=12.0.0"
  },
  "scripts": {
    "clean": "uu_appg01_devkit clean",
    "test": "uu_appg01_devkit test",
    "build": "node tools/prebuild.js && uu_appg01_devkit build",
    "package": "uu_appg01_devkit package",
    "dist": "uu_appg01_devkit package",
    "start": "uu_appg01_devkit start",
    "prettify": "uu_appg01_devkit prettify",
    "vulnerabilityScan": "uu_appg01_devkit vulnerabilityScan",
    "uuAppDevKitUpgrade": "uu_appg01_devkit uuAppDevKitUpgrade",
    "uu5Generate": "uu5devkitg01-plugin generate",
    "uu5LsiExport": "uu5devkitg01-plugin lsiExport",
    "uu5LsiImport": "uu5devkitg01-plugin lsiImport",
    "uu5DocUpdateExamples": "uu5devkitg01-plugin docUpdateExamples"
  },
  "main": "src/index.js",
  "files": [
    "test/setup",
    "dist",
    "dist-node",
    "jsx"
  ],
  "prettier": {
    "printWidth": 120
  },
  "jest": {
    "transformIgnorePatterns": [
      "^(?!.*/node_modules/uu_appg01).*/node_modules/",
      "uu5g05-test"
    ]
  }
}
