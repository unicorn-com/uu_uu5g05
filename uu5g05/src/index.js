import "./_internal/polyfills/polyfills.js";

import * as Exports from "./exports.js";
import LibraryRegistry from "./utils/library-registry.js";
import "./_internal/startup.js";

export * from "./exports.js";
export default Exports;

if (process.env.NODE_ENV !== "test") {
  console.log(
    `${process.env.NAME}-${process.env.VERSION} Â© Unicorn\nTerms of Use: https://unicorn.com/tou/${process.env.NAME}`,
  );
}
LibraryRegistry.registerLibrary({
  name: process.env.NAME,
  version: process.env.VERSION,
  namespace: process.env.NAMESPACE,
});

// we'll still export uu5g05 into global variable so that JSX pragmas can be used
// (Uu5g05.Utils.Element.create + Uu5g05.Fragment), however, we'll prevent from adding
// anything to the value
// eslint-disable-next-line no-import-assign
const frozenExports = Object.freeze(Exports);
const publish = (scope, exports) => {
  scope.Uu5 = exports;
  scope.Uu5g05 = {
    Utils: {
      Element: {
        create: exports.Utils.Element.create,
      },
    },
  };
  Object.defineProperty(scope.Uu5g05, "Fragment", Object.getOwnPropertyDescriptor(exports, "Fragment"));
};
if (typeof window !== "undefined") publish(window, frozenExports);
if (process.env.NODE_ENV === "test" && typeof global !== "undefined") publish(global, frozenExports); // Jest tests
