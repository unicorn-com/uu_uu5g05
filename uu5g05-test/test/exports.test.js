import { Utils } from "uu5g05";
import Exports from "uu5g05-test";

const memo = Utils.Component.memo(() => {});

describe(`Uu5`, () => {
  it("exported API", async () => {
    let apiMap = {};
    addKeys(Exports, apiMap);
    expect(apiMap).toMatchSnapshot();
  });
});

function addKeys(obj, result, filter = (key, obj, prefix) => true, prefix = "", addedSet = new Set()) {
  if (prefix) result[prefix] = typeof obj;
  if (!obj || (typeof obj !== "object" && typeof obj !== "function")) return;
  if (addedSet.has(obj)) return;
  addedSet.add(obj);
  for (let k of Object.getOwnPropertyNames(obj)) {
    if (/^_/.test(k)) continue;
    if (
      /^(defaultProps|displayName|isStateless|isUu5PureComponent|length|logger|name|nestingLevel|propTypes|prototype|uu5ComponentType|uu5Tag|getDerivedStateFromError|getDerivedStateFromProps)$/.test(
        k,
      )
    ) {
      continue;
    }
    if (obj.$$typeof === memo.$$typeof && k in memo) continue;
    if (!filter(k, obj, prefix)) continue;
    let childFilter;
    if (/^[a-z]/.test(k) || /^[A-Z_0-9]+$/.test(k)) childFilter = () => false;
    else if (/^[A-Z]/.test(k)) childFilter = (k, obj) => k in obj;
    else continue;
    addKeys(obj[k], result, childFilter, prefix ? prefix + "." + k : k, addedSet);
  }
}
