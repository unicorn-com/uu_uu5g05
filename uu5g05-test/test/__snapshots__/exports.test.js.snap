// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uu5 exported API 1`] = `
{
  "Config": "object",
  "Config.Css": "object",
  "Config.Css.cache": "object",
  "Config.Css.css": "function",
  "Config.Css.cx": "function",
  "Config.Css.flush": "function",
  "Config.Css.getRegisteredStyles": "function",
  "Config.Css.hydrate": "function",
  "Config.Css.injectGlobal": "function",
  "Config.Css.join": "function",
  "Config.Css.keyframes": "function",
  "Config.Css.merge": "function",
  "Config.Css.sheet": "object",
  "Config.TAG": "string",
  "FormInputComponent": "object",
  "FormInputComponent.setup": "function",
  "FormInputComponent.testExtensionInputProperties": "function",
  "FormInputComponent.testProperties": "function",
  "FormItemComponent": "object",
  "FormItemComponent.setup": "function",
  "InputComponent": "object",
  "InputComponent.setup": "function",
  "InputComponent.testProperties": "function",
  "Session": "object",
  "Session.TEST_IDENTITY": "object",
  "Session.instance": "object",
  "Session.reset": "function",
  "Session.setExpiring": "function",
  "Session.setIdentity": "function",
  "Session.setPending": "function",
  "Test": "object",
  "Test.Wrapper": "function",
  "Test.act": "function",
  "Test.buildQueries": "function",
  "Test.cleanUp": "undefined",
  "Test.configure": "function",
  "Test.createEvent": "function",
  "Test.createHookComponent": "function",
  "Test.fireEvent": "function",
  "Test.getNodeText": "function",
  "Test.getRoles": "function",
  "Test.isInaccessible": "function",
  "Test.logRoles": "function",
  "Test.prettyDOM": "function",
  "Test.queries": "object",
  "Test.queryHelpers": "object",
  "Test.render": "function",
  "Test.renderHook": "function",
  "Test.screen": "object",
  "Test.userEvent": "object",
  "Test.waitFor": "function",
  "Test.waitForElementToBeRemoved": "function",
  "Test.within": "function",
  "Utils": "object",
  "Utils.omitConsoleLogs": "function",
  "Utils.wait": "function",
  "VisualComponent": "object",
  "VisualComponent.colorConvert": "function",
  "VisualComponent.setup": "function",
  "VisualComponent.testBackground": "function",
  "VisualComponent.testBorderRadius": "function",
  "VisualComponent.testColorScheme": "function",
  "VisualComponent.testProperties": "function",
  "VisualComponent.testSignificance": "function",
  "VisualComponent.testTooltip": "function",
  "VisualComponent.testTooltipLsi": "function",
  "act": "function",
  "initHookRenderer": "function",
  "mount": "function",
  "omitConsoleLogs": "function",
  "renderHook": "function",
  "setInputValue": "function",
  "shallow": "function",
  "testProperties": "function",
  "testProperty": "function",
  "wait": "function",
  "waitUntilCalled": "function",
  "waitUntilCalledTimes": "function",
  "waitWhile": "function",
}
`;
